code/weapon/weapons.cpp

```diff
@@ -8812,3 +8812,91 @@ bool weapon_has_iff_restrictions(weapon_info* wip)
 			return restriction.first == LockRestrictionType::IFF; 
 		});
 }
+
+bool weapon_secondary_world_pos_in_range(object* shooter, weapon_info* wip, vec3d* target_world_pos)
+{
+	vec3d vec_to_target;
+	vm_vec_sub(&vec_to_target, target_world_pos, &shooter->pos);
+	float dist_to_target = vm_vec_mag(&vec_to_target);
+
+	float weapon_range;
+	//local ssms are always in range :)
+	if (wip->wi_flags[Weapon::Info_Flags::Local_ssm])
+		weapon_range = wip->lssm_lock_range;
+	else
+		// if the weapon can actually hit the target
+		weapon_range = MIN((wip->max_speed * wip->lifetime), wip->weapon_range);
+
+
+	extern int Nebula_sec_range;
+	// reduce firing range in nebula
+	if ((The_mission.flags[Mission::Mission_Flags::Fullneb]) && Nebula_sec_range) {
+		weapon_range *= 0.8f;
+	}
+
+	return dist_to_target <= weapon_range;
+}
+
+bool weapon_multilock_can_lock_on_subsys(object* shooter, object* target, ship_subsys* target_subsys, weapon_info* wip, float* out_dot) {
```

Comment:
Remove "multilock" from the function name since it should be applicable to single-lock missiles too.

Reply:
While all of this is indeed checked for single-target missiles, these functions *are* only used by multilock, which I think is important information to convey. I guess I would say that the "multilock" here is less about what its actually doing and instead about who uses these functions.

Reply:
Oh hm.  Yes, that is important information to convey, but I thought these functions were used by single-target also.

Reply:
They still do all these things but they are a lot more spread around, its doing a bunch of different stuff in between all these checks. These functions are basically optimized to only the bare minimum for checking the validity of a target (since its being done en masse). If you still think the name should be changed, fair enough, but that's the situation.

Reply:
Gotcha.  Then keep the current name, but add a comment to document that this function really is only used by multilock.